### PR TITLE
feat: Integrate fastlane and code push to appcenter

### PR DESCRIPTION
**Pull request by tuledu**

### CODE REVIEW

File: template/android/fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    appcenter_codepush(
+      env: options[:env],
+      owner_name: owner_name,
+      my_app: my_app,
+      platform: platform
+    )
+  end
+
+  desc "Build or codepush" #type = [codepush, build]
+  lane :testif do |options|
```
Comment (TranQuangPhi) [inline template/android/fastlane/Fastfile:65]: replace "testif" to "build"

File: template/.husky/pre-commit [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-yarn pretty && yarn lint:fix
+cd template && yarn pretty && yarn lint:fix
```
Comment (ThinhKimVo) [inline template/.husky/pre-commit:4]: remove

**Review by ThinhKimVo — APPROVED** [review]

File: docs/fastlane.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Enter file ~/.bash_profile
+``vi ~/.bash_profile``
+Use the export command to add new environment variables
+``export APPLE_ID=your_apple_id``
+Execute the new ~/.bash_profile by either restarting the terminal window or using:
+``source ~/.bash_profile``
+This way only works after you run ``source ~/.bash_profile`` when open terminal
+You don't need to run it when do this way:
+If you use zshrc then:
+put ``source ~/.bash_profile`` in ``~/.zshrc``
```
Comment (ThinhKimVo) [inline docs/fastlane.md:16]: edit
```Add the following lines to your ~/.zprofile or ~/.zshrc (if you are using bash, then ~/.bash_profile or ~/.bashrc) config file:```

File: template/android/fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+default_platform(:android)
+
+platform :android do
+
+  owner_name = "Saigontechnology" #Replace owner name here
+  my_app = "ReactNativeBaseAndroid" #Replace my app here
+  platform = 'android'
+
+  desc "Runs all the tests"
```
[the remark this answers is not in the document]
Comment (tuledu) [reply template/android/fastlane/Fastfile]: Done

File: template/android/fastlane/Fastfile (hunk 1 above)
Comment (tuledu) [reply template/android/fastlane/Fastfile]: Done

File: template/.husky/pre-commit (hunk 2 above)
Comment (tuledu) [reply template/.husky/pre-commit]: Done

File: docs/fastlane.md (hunk 3 above)
Comment (tuledu) [reply docs/fastlane.md]: Done

**Review by chungnguyenhoang — APPROVED** [review]

**Review by ThinhKimVo — APPROVED** [review]

**Review by anhle10051996 — APPROVED** [review]

**Pull request merged**